wfdispatcher/objects/workflowmanager.py

```diff
@@ -132,9 +128,11 @@ def define_workflow(self, data):
             lm = LSSTMiddleManager(parent=self,
                                    config=LSSTConfig(),
                                    user=self.user,
-                                   spawner=MockSpawner(parent=self),
-                                   authenticator=self.auth)
-            lm.spawner.user = self.user
+                                   authenticator=AM(parent=self))
+            lm.authenticator.set_auth_fields(self.user)
+            # That created a spawner on the authenticator with correct
+            #  fields.
+            lm.spawner = lm.authenticator.spawner
```

Comment:
I think this is more of a nit, so approving anyway, but this seems weirdly complicated to me and I'm not sure why the authenticator should be responsible for creating a spawner with the information that the middle manager already has. Why not have the middle manager create the spawner directly, since it already has all the required information to do so?

Reply:
Oh, that's easy.  In the jupyterhub world, the Middle Manager is created by the Spawner.